### PR TITLE
feat(component): `<VideoFeature />`

### PR DESCRIPTION
**Pull request by EnMod**
🎟️ [Asana Task](https://app.asana.com/0/0/1203253460638593/f)
🔍 [Preview Link](https://react-components-git-nqvf-build-component-hashicorp.vercel.app/component/videofeature)

---

## Description

In this PR, the new `VideoFeature` component will be built out:
- [x] Props document updated
- [x] `docs.mdx` updated to reflect component props and showcase the component
- [x] Tests developed just enough to pass; further testing work will be handled in a future PR merged to assembly

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/video-feature/types.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import type { IntroProps } from '@hashicorp/react-intro/types'
+import type { InlineVideoProps } from '@hashicorp/react-inline-video/types'
+
+export interface VideoFeatureProps {
+  appearance?: 'light' | 'dark'
+  contentPosition?: 'left' | 'right'
+  heading: IntroProps['heading']
+  description: IntroProps['description']
+  author?: AuthorBylineProps
+  video: InlineVideoProps
```
Comment (alexcarpenter) [inline packages/video-feature/types.ts:11]: Should we be picking the props we need here vs passing all of them along?

Wouldn'nt assume we expose appearance for author or video as those come from video.

Video also does not need to expose the gradientPosition as that is inferred from the contentPosition etc.

File: packages/video-feature/style.module.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
+.videoFeature {
+  --padding-y: 48px;
+  --padding-x: 48px;
+  --background-color: var(--wpl-neutral-100);
+
+  padding: var(--padding-y) var(--padding-x);
+  background-color: var(--background-color);
+  display: flex;
+  flex-direction: column-reverse;
```
Comment (alexcarpenter) [inline packages/video-feature/style.module.css:9]: flex usage is causing some layout issues here. Can we use grid?

File: packages/video-feature/types.ts (hunk 1 above)
Comment (EnMod) [reply packages/video-feature/types.ts]: Good points all around! I've used `Pick` instead to more specifically assign the type here:
https://github.com/hashicorp/react-components/pull/784/commits/98fe500d8415fff17d0f358ebb77b764e814ba0d

File: packages/video-feature/style.module.css (hunk 2 above)
Comment (EnMod) [reply packages/video-feature/style.module.css]: Update: I've refactored the layout from `flex` to `grid` here. Some nice gains in readability, without the sizing/layout issues from the `flex` layout:
https://github.com/hashicorp/react-components/pull/784/commits/8cc180319ce97c319fada6d4e3184c641d068323

File: packages/video-feature/index.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  author,
+  video,
+}: VideoFeatureProps) {
+  return (
+    <div
+      className={classNames(s.videoFeature, s[appearance], s[contentPosition])}
+    >
+      <div className={s.content}>
+        <Intro
+          heading={heading}
```
Comment (alexcarpenter) [inline packages/video-feature/index.tsx:22]:
```suggestion
          heading={heading}
          headingSize={3}
```

Comment (EnMod) [reply packages/video-feature/index.tsx:22]: Good catch!

**Comment by EnMod** [comment]
@alexcarpenter I've tackled tests in a separate PR (#794). Plans going forward for this component:

1. Merge to the `<VideoFeature />` assembly when approved (#783 , it autoclosed after a rebase, please disregard)
2. Rebase and merge the tests PR (#794) into assembly when approved
3. Add the changeset in assembly

**Review by alexcarpenter — APPROVED** [review]

**Pull request merged**